### PR TITLE
Attach index name in error message

### LINKED ISSUE #4392: Error message could be clearer
**Describe the bug**

     "Attribute `id` is not sortable. This index does not have configured sortable attributes.",

**Expected behavior**

I would LOVE to see the name of the index in this message.

     "Attribute `id` is not sortable. index 'Author' does not have configured sortable attributes.",

In particular, I _do_ have sortable attributes:

http://127.0.0.1:7700/indexes/Author/settings

image

Perhaps the real issue is that the id is the primary key?  More likely I'm using the wrong index, but I can't see what that is.  A link to the settings would be even more awesome!

### PR DESCRIPTION
**Pull request by airycanon**
# Pull Request

## Related issue
Fixes #4392 

## What does this PR do?
- ...

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by airycanon** [comment]
@irevoire could you please take a look at this PR?

**Comment by airycanon** [comment]
kindly ping @irevoire

**Comment by Kerollmops** [comment]
Hey @airycanon 👋

@irevoire is on vacation and will come back next week. He will dedicate some time to review your PR soon.

Thank you for your contribution 😊

File: crates/index-scheduler/src/batch.rs [hunk 1]
```diff
@@ -781,7 +780,9 @@ impl IndexScheduler {
                     let index = self.index_mapper.index(&rtxn, name)?;
                     let dst = temp_snapshot_dir.path().join("indexes").join(uuid.to_string());
                     fs::create_dir_all(&dst)?;
-                    index.copy_to_file(dst.join("data.mdb"), CompactionOption::Enabled)?;
+                    index
+                        .copy_to_file(dst.join("data.mdb"), CompactionOption::Enabled)
+                        .map_err(|e| Error::from_milli(e, Some(uuid.to_string())))?;
```
Comment (irevoire) [inline crates/index-scheduler/src/batch.rs:785]:
```suggestion
                        .map_err(|e| Error::from_milli(e, Some(name.to_string())))?;
```

here it's the name of the index that we want to return

**Review by irevoire — CHANGES_REQUESTED** [review]
Hey, the implementation looks really good!
I think we'll be able to merge this PR in no time, thanks a lot and sorry for the delay :pray: 

I left a few comments here and there, but nothing hard to fix.
And could you rebase on the `release-v1.12` branch because that's definitely something I would like to have for the next release.
I'll update the base branch on github directly.

**Comment by airycanon** [comment]
@irevoire Thanks for you guidance, I've addressed all the comments in this PR. Could you please take another look?

**Review by irevoire — APPROVED** [review]
Hey @airycanon,

Looks awesome! Thanks for your PR 🫶 
Merging
bors merge

**Pull request merged**